### PR TITLE
doc: Intelligent Peer Sharing description

### PR DESCRIPTION
**Pull request by asmie**
Description of the IPS.

### CODE REVIEW

**Comment by kylegranger** [comment]
I'm editing the document separately -- easier that way (for this case).

File: doc/ips.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
++ eigenvector – this tells how much influence node's neighbours have.
++ closeness – this measure calculates the shortest paths between all nodes, then assigns each node a score based on its sum of shortest paths. This is not very relevant here, as neither the density or sparseness of a network is intrinsically bad. IPS tries to keep its own centrality high and connect to peers with high closeness (if the MCDA weights allow it).
+
+All of the above factors are written to the IPS result log, allowing one to check how a particular run influenced the network: the state before may be compared to the the state afterwards.
+
+### Security checks
+
+One of the most important properties of a network topology is the presence or absence of network islands. Presence of islands may influence every other network or node parameter that we are observing. Merging two massive islands can be risky and should not be done automatically. The islands could have been disconnected for a long time and produced different history of their blockchain. IPS can detect such situation and breaks computations notifying the user about island existence.
```
Comment (kylegranger) [inline doc/ips.md:30]: `a different history`
`IPS can detect such situations, and notify the user about the existence of islands` : just keep it simple here

File: doc/ips.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+All of the above factors are written to the IPS result log, allowing one to check how a particular run influenced the network: the state before may be compared to the the state afterwards.
+
+### Security checks
+
+One of the most important properties of a network topology is the presence or absence of network islands. Presence of islands may influence every other network or node parameter that we are observing. Merging two massive islands can be risky and should not be done automatically. The islands could have been disconnected for a long time and produced different history of their blockchain. IPS can detect such situation and breaks computations notifying the user about island existence. 
+
+Next, IPS checks if the network can be easily fragmented by attacking a given percent of the nodes and if so, preventing such cases by creating new connections between their neighbours. Nodes selected to that fragmentation simulation are chosen from the "hot" nodes, which means the nodes with highest betweenness factor.
```
Comment (kylegranger) [inline doc/ips.md:32]: `Nodes selected by that fragmentation simulation`

**Review by kylegranger — APPROVED** [review]
:Thu

File: doc/ips.md (hunk 1 above)
Comment (asmie) [reply doc/ips.md]: 👍

File: doc/ips.md (hunk 2 above)
Comment (asmie) [reply doc/ips.md]: 👍

**Pull request merged**